kubernetes/update.go

```diff
@@ -141,7 +141,19 @@ func updateDeployment(ctx context.Context, deployCli v1.DeploymentInterface, dep
 		Str("deploymentName", deploymentName).
 		Msg("Waiting for deployment to complete")
 
-	return waitForDeployment(ctx, deployCli, newDeployment.Name, newDeployment.UID)
+	err = nil
+	numRetries := 1
+	for numRetries < 10 {
+		err = waitForDeployment(ctx, deployCli, newDeployment.Name, newDeployment.UID)
```

Comment:
Do you really need to retry the waitForDeployment or just increase the timeout?  It's a watch afterall and will return as soon as the deployment has completed.  We should give up after 5 minutes.   I don't think a retry makes sense personally unless you couldn't create the watch.

I'd make the timeout a parameter to `waitForDeployment`.  I don't see the point of re-creating a watch process personally.

There is also a 5s delay in the waitForDeployment function.   What's that about?  Is it sufficient?

Reply:
Got your point. Reverted the code